src/eth_provider/database/ethereum.rs

```diff
@@ -105,6 +107,14 @@ pub trait EthereumBlockStore {
 
 #[async_trait]
 impl EthereumBlockStore for Database {
+    async fn latest_header(&self) -> Result<Option<Header>, EthApiError> {
+        let filter = EthDatabaseFilterBuilder::<filter::Header>::default().build();
+        Ok(self
+            .get_one::<StoredHeader>(filter, Some(doc! { "number": -1 }))
```

Comment:
Here why not using our FindOpts wrapper with a `with_sort` function:

https://github.com/mongodb/mongo-rust-driver/blob/b781af26dfb17fe62823a866a025de9fb102e0b3/src/coll/options.rs#L845

This probably also requires to modify the `get_one` function signature and adding one function to our wrapper

Reply:
Yes but I think you should be able to build from https://github.com/kkrt-labs/kakarot-rpc/blob/4003bd0518102f7cd971d86afa143f86253f6e07/src/eth_provider/database/mod.rs#L43